doc/guides/sample_app_ug/cnet_graph.rst

```diff
@@ -0,0 +1,286 @@
+..  SPDX-License-Identifier: BSD-3-Clause
+    Copyright (c) 2020 Marvell International Ltd.
+    Copyright (c) 2022 Red Hat, Inc.
+
+CNET Graph Sample Application
+=============================
+
+The CNET graph sample application demonstrated how to use a set of graph nodes
+(`cnet stack library`_) to create a UDP/TCP network stack that uses channel
+callbacks in user space to process TCP and UDP traffic.
+
+Overview
+--------
+The application demonstrates the use of the graph framework and graph nodes:
+
+- ``chnl_callback``
+- ``chnl_recv``
+- ``arp_request``
+- ``eth_rx``
+- ``eth_tx``
+- ``ptype``
+- ``ip4_input``
+- ``ip4_output``
+- ``ip4_forward``
+- ``ip4_proto``
+- ``punt_kernel``
+- ``kernel_recv``
+- ``gtpu_input``
+- ``tcp_input``
+- ``tcp_output``
+- ``udp_input``
+- ``udp_output``
+- ``null``
+- ``pkt_drop``
+- ``eth_rx-X`` (where X is the port id)
+- ``eth_tx-X`` (where X is the port id)
+
+
+Running the Application
+-----------------------
+
+Edit the cnetfwd-graph.jsonc file to change the configuration, then run::
+
+    sudo ./builddir/examples/cnet-graph/cnet-graph -c <json-file-name>
+
+.. _cnet_graph_explanation:
+
+Explanation
+-----------
+
+The following sections describe the details of the CNET Graph application.
+
+Graph Node Pre-Init Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+After device configuration is complete, the lport ids are passed to ``cnet_eth_node_config()``.
+This causes the ``eth_rx`` and ``eth_tx`` nodes to be cloned as ``eth_rx-X`` and ``eth_tx-X``
+where X represents the port id associated with the node. For the ``eth_rx-X`` node, this
+function also clones the edges of ``eth_rx``. This function sets the ``eth_tx-X`` node as the
+next node after the ``ip4_forward`` and ``ip4_output``.
+
+.. code-block:: c
+
+    static struct pkt_eth_node_config pkt_conf[CNE_MAX_ETHPORTS];
+    static int
+        initialize(void)
```

Comment:
super small nit, the indent is off on this and it shows up in the rendered document as well.

Reply:
https://github.com/CloudNativeDataPlane/cndp/pull/242